client/src/pages/Chart.tsx

```diff
@@ -0,0 +1,251 @@
+import { useEffect, useRef, useState } from 'react';
+import { Card, CardContent, Typography, Box, Select, MenuItem, FormControl, InputLabel } from '@mui/material';
+import PubNub from 'pubnub';
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+  Filler,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+import { PUBNUB_CHANNELS, TIME_RANGES } from '@mevscan/shared/constants';
+import './Chart.css';
+import { GrossMevDataResponse } from '@mevscan/shared';
+
+// Register Chart.js components
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend,
+  Filler
+);
+
+function Chart() {
+  const pubnubRef = useRef<PubNub | null>(null);
+  const [grossMevData, setGrossMevData] = useState<GrossMevDataResponse[]>([]);
+  const [timeRange, setTimeRange] = useState<string>('15min');
+
+  // Helper function to format time range label
+  const formatTimeRangeLabel = (range: string): string => {
+    if (range === '5min') return '5 min';
+    if (range === '15min') return '15 min';
+    if (range === '30min') return '30 min';
+    if (range === '1hour') return '1 hour';
+    if (range === '12hours') return '12 hours';
+    return range;
+  };
+
+  
+  // Initialize PubNub and subscribe to channel
+  useEffect(() => {
```

Comment:
Can you check if there is a pubnub react hook? however I'd prefer to define a custom hook instead of useEffect